### ISSUE TITLE
Send breaks if body contains multi-byte utf-8 characters

### ISSUE THREAD

**Issue by klemetsson** [rabbitmq/rabbitmq-web-stomp-examples#2]
In **stomp.js**, the Frame.prototype.toString function sets the content-length to the number of characters in the body. This is only valid if there is one byte per character.

I added the following function to calculate the number of bytes in the body:

```javascript
function countBytes(string) {
  var escapedStr = encodeURI(string);
  if (escapedStr.indexOf("%") != -1) {
    var count = escapedStr.split("%").length - 1;
    if (count == 0) count++;
    var tmp = escapedStr.length - (count * 3);
    return count + tmp;
  }
  else return escapedStr.length;
}
```

and changed

```javascript
lines.push("content-length:" + ('' + this.body).length);
```

to

```javascript
lines.push("content-length:" + countBytes('' + this.body));
```

Maybe there is a cleaner way of doing this. But this change worked for me.
There is possibly more issues of this type in stomp.js. The .length gives the number of characters and not the number of bytes in a string.

**Comment by michaelklishin** [comment]
@essen is this still relevant in `3.6.x` and `master`?

**Comment by essen** [comment]
I'll check.

**Comment by essen** [comment]
It's been fixed upstream. Perhaps we should just update the stomp.js file.

On a related note, upstream (https://github.com/jmesnil/stomp-websocket) is not maintained anymore. The author seemed interested to donate the project to ActiveMQ (http://activemq.2283324.n4.nabble.com/Code-donation-for-stomp-js-td4694260.html). Either way we can probably move to the latest version for now and deal with that later if needed.

**Comment by michaelklishin** [comment]
Yes, I think we should upgrade and see what's going to happen with the JS library.

**Comment by michaelklishin** [comment]
Fixed in #10 by @essen. Thank you!
